google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolOptionsTest.java

```diff
@@ -50,16 +49,26 @@
   @Test
   public void setMinMaxSessions() {
     if (minSessions > maxSessions) {
-      expectedException.expect(IllegalArgumentException.class);
-    }
-    SessionPoolOptions options =
-        SessionPoolOptions.newBuilder()
-            .setMinSessions(minSessions)
-            .setMaxSessions(maxSessions)
-            .build();
+      try {
+        SessionPoolOptions options =
+            SessionPoolOptions.newBuilder()
+                .setMinSessions(minSessions)
+                .setMaxSessions(maxSessions)
+                .build();
+        fail("Expected exception");
+      } catch (IllegalArgumentException ex) {
+        assertNotNull(ex.getMessage());
+      }
+    } else {
+      SessionPoolOptions options =
+          SessionPoolOptions.newBuilder()
+              .setMinSessions(minSessions)
+              .setMaxSessions(maxSessions)
+              .build();
 
-    assertThat(minSessions).isEqualTo(options.getMinSessions());
-    assertThat(maxSessions).isEqualTo(options.getMaxSessions());
+      assertThat(minSessions).isEqualTo(options.getMinSessions());
+      assertThat(maxSessions).isEqualTo(options.getMaxSessions());
+    }
```

Comment:
I'm not a huge fan of this repetition. Can we rewrite it as follows?
```
try {
  SessionPoolOptions options =
      SessionPoolOptions.newBuilder()
          .setMinSessions(minSessions)
          .setMaxSessions(maxSessions)
          .build();

  if (minSessions > maxSessions) {
    fail("Expected exception");
  }

  assertThat(minSessions).isEqualTo(options.getMinSessions());
  assertThat(maxSessions).isEqualTo(options.getMaxSessions());
} catch (IllegalArgumentException ex) {
  if (minSessions <= maxSessions) {
    throw ex;
  }

  assertNotNull(ex.getMessage());
}
```

Reply:
Done.